src/libopenrave/kinbody.cpp

```diff
@@ -6395,10 +6390,9 @@ UpdateFromInfoResult KinBody::UpdateFromKinBodyInfo(const KinBodyInfo& info)
     }
 
     // transform
-    if( info.IsModifiedField(KinBodyInfo::KBIF_Transform) || (info._vLinkInfos.size() > 0 && info._vLinkInfos[0]->IsModifiedField(KinBody::LinkInfo::LIF_Transform)) ) {
-        Transform bodyTransform = info._transform * _invBaseLinkInBodyTransform;
-        if( GetTransform().CompareTransform(bodyTransform, g_fEpsilon) ) {
-            SetTransform(bodyTransform);
+    if( info.IsModifiedField(KinBodyInfo::KBIF_Transform) ) {
+        if( GetTransform().CompareTransform(info._transform, g_fEpsilon) ) {
```

Comment:
This if-statement is actually the only change that this pull request needs. Here we should be comparing `GetTransform()` (`this` kinbody's transform) with `info._transform` (the transform of the input kinbody).

Also, I think `info._vLinkInfos.size() > 0 ...` check above should not be removed.

Reply:
The check should be removed simply because the internal state will not change whether the input `info` changed at this point.